src/test/java/seedu/address/model/task/DeadlineTaskTest.java

```diff
@@ -43,9 +43,12 @@ public void equals_isEqual_returnsTrue() throws Exception {
 
     @Test
     public void equals_notEqual_returnsFalse() throws Exception {
-        final DeadlineTask other = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(2));
-        assertFalse(deadlineTask.equals(other));
-        assertFalse(deadlineTask.hashCode() == other.hashCode());
+        final DeadlineTask other1 = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(2), false);
+        final DeadlineTask other2 = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(1), true);
+        assertFalse(deadlineTask.equals(other1));
+        assertFalse(deadlineTask.hashCode() == other1.hashCode());
+        assertFalse(deadlineTask.equals(other2));
+        assertFalse(deadlineTask.hashCode() == other2.hashCode());
```

Comment:
Yeah, this is definitely an improvement.

I think the testing for `equals()` that we do needs to be more comprehensive though (not needed for this PR). Then again, testing for all possibilities of all the fields is a lot of work. I wonder if we could use something like [EqualsVerifier](http://jqno.nl/equalsverifier/) for this.